### PR TITLE
Floorbot bug fix + Bridgemode Removal

### LINKED ISSUE #22472: Floorbots do not work anymore
**Issue Description**:
Floorbots, when trying to fix a damaged floor tile, will stay in the place doing their animation, will eat through their entire stack but will not remove the actual floor tile, leaving it unfixed.
I have watched a floorbot eating through its stack on the same tile for 10 minutes, it is sad

**What did you expect to happen**: 
Floorbot to remove the damage floor tile, then replace it.

**What happened instead**:
Floorbot did not remove damage floor tile, instead placed *ghost* tiles, basically voiding its entire inventory and metal

**Why is this bad/What are the consequences**:
1. Floorbot doesn't function properly
2. It will just eat through its stacks without doing anything, and thus it will suck metal from station

**Steps to reproduce the problem**:
1. Damage a floor
2. Get floorbot there
3. watch in agony

**When did the problem start happening**:
Absolutely no idea, people do not use them much

**Extra information**:

### LINKED ISSUE #22478: Floorbot is using up tiles when "removing" tile (makes no sense)
**Issue Description**:
Floorbot is using up tiles when removing damaged tile (it don't even replace it with plating)

**What did you expect to happen**: 
remove damaged floor and dont consume tiles for it (its just crowbar action)

**What happened instead**:
removed tile, used up tiles

**Why is this bad/What are the consequences**:
it consume tiles on action with should not use up tiles, it makes no sense.

**Steps to reproduce the problem**:
1. Make Floorbot.
2. Give it tiles.
3. Enable "repair damaged tiles and platings"
4. Look Floorbot use up tiles to remove damaged tile.

**When did the problem start happening**:
No idea when

**Extra information**:
It make no sense to use up tiles when it normally when you do it dont require tiles to do so.

### PR DESCRIPTION
**Pull request by Oyu-De**
## What Does This PR Do
Fixes #22472
Fixes #22478 
Fixes floorbot eating up tiles nonstop.
Removes bridge mode from floorbot

## Why It's Good For The Game
Floorbot usage is rare, and discussed with people randomly and seems like, floorbot in on itself is not used much, but bridge mode function isnt even known in general. Removes this old and ancient mode/option

## Testing
Flame Thrower an area, remove some tiles and space some. Watch the bot suffer
image

## Changelog
:cl:
fix: No longer gets stuck and eats up tiles, actually does place a tile
del: Removed Bridge Mode option
/:cl:

### CODE REVIEW

**Comment by Oyu-De** [comment]
Ok small update. this partially fixes it. It fixes the replace tile and autotile interaction, while the broken floor tile and autotile interaction is still broken. This is a bridgemode removal + partial bug fix. I am still testing the autotile and broken/damaged floor tile.

**Comment by Oyu-De** [comment]
Ok small update. this partially fixes it. It fixes the replace tile and autotile interaction, while the broken floor tile and autotile interaction is still broken. This is a bridgemode removal + partial bug fix. I am still testing the autotile and broken/damaged floor tile.

And due to that, putting this on hold from my end

**Comment by Oyu-De** [comment]
Ok, so, fixed it with adding a `F.break_tile_to_plating()` for autotile check
should cover the problems, tested and seems to be working fine without interfering with other behaviours

**Comment by Oyu-De** [comment]
> Ok, so, fixed it with adding a `F.break_tile_to_plating()` for autotile check should cover the problems, tested and seems to be working fine without interfering with other behaviours

Alright update on this, may have one last oddity. after that it should be all set

**Comment by Oyu-De** [comment]
Ready for review

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -317,7 +280,7 @@
 	anchored = TRUE
 
 	if(isspaceturf(target_turf)) //If we are fixing an area not part of pure space, it is
-		visible_message("<span class='notice'>[targetdirection ? "[src] begins installing a bridge plating." : "[src] begins to repair the hole."] </span>")
+		visible_message("<span class='notice'>["[src] begins to repair the hole."] </span>")
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:283]:
```suggestion
		visible_message("<span class='notice'>[src] begins to repair the hole.</span>")
```

Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm]: commiting, but for learning purposes, what is the pupose of the change

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 2]
```diff
@@ -343,13 +306,19 @@
...
-	if(autotile) //Build the floor and include a tile.
+	if(replacetiles)
+		F.break_tile_to_plating()
 		target_turf.ChangeTurf(/turf/simulated/floor/plasteel)
-	else //Build a hull plating without a floor tile.
-		target_turf.ChangeTurf(/turf/simulated/floor/plating)
+	else
+		if(autotile) //Build the floor and include a tile.
+			F.break_tile_to_plating()
+			target_turf.ChangeTurf(/turf/simulated/floor/plasteel)
```
[the remark this answers is not in the document]
Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:319]: I am unsure which part you are referring to

[the remark this answers is not in the document]
Comment (Henri215) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:319]: both replacetiles and autotile do break tile to plating and change the turf to plasteel floor, i dont remember much about the floorbots mode and dont know exactly what each one should do.

[the remark this answers is not in the document]
Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:319]: sorta, replace tiles main function is to place NEW floor tiles where there is none, and autotile is to *FIX*, aka remove the broken one and place a new one while fixing the plating. replace tiles also tries to remove the tile when repairing if it is on iirc, (gonna double check this to make sure).
They both have their seperate function, but while repairing they both will require the break_tile call

[the remark this answers is not in the document]
Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:319]: update: having replace tiles on while fixing does fix the tiles just like autotile, so it needs replace tiles. Just that the main difference is that replace tile is to add flooring to anything and auto tile is to just repair floor tiles

[the remark this answers is not in the document]
Comment (Henri215) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:319]: Got it now, you can resolve this.

**Review by lewcc — APPROVED** [review]

**Pull request merged**